### DISCUSSION TITLE
Best way to integrate Single File? + A glitch?

### DISCUSSION THREAD

**Discussion by anishchand99** [gildas-lormeau/SingleFile#1274] [Q&A]
First of all, thanks the community for a great tool. I am currently using single file as I have the need to save several pages offline preserving all the details (as a personal tool). Some of the pages needed to be scrolled to bottom to ensure every item is loaded. So currently I am using puppeteer to visit the page and scroll to the bottom. 
- I tried to inject the minified js scripts into page which makes the singlefile.getPageData() available. It is easy to integrate this way but I found out that the iframes are empty, unlike the chrome extension that preserves them correctly.
- So the next method, I tried was downloading the repo and using as a chrome extension for puppeteer and sending the 'save' message to the extension whenever I need to download the page.

Are there any other ways to integrate SingleFile to puppeteer (so I can test if the results are better/worse)?

Also, I am not fully sure if this is a bug/issue. 
Sometimes singlefile when downloading the page gets stuck on the green box with tree white dots and seems to freeze and not download the page. <img width="41" alt="Screenshot 2023-10-16 at 11 09 08 PM" src="https://github.com/gildas-lormeau/SingleFile/assets/29545745/78145b41-1aec-4032-8c1a-19eb7e7d4272">

Is this a known issue or maybe a mistake on my part. It can be observed when I am using the extension from chrome store and manually browsing the web. And also when I use the extension with puppeteer. I am not sure when it happens. But for example: on ```https://www.washingtonpost.com/``` when I click the button to download, it gets stuck on green. When I reload the page, then the extension automatically gets triggered and saves the page.

**Comment by gildas-lormeau** [comment]
Did you take a look at SingleFile CLI? See https://github.com/gildas-lormeau/single-file-cli. This program uses (by default) puppeteer to save pages with SingleFile.

Regarding the "three white dots issue", this looks like a bug with script injection. This can be seen in the background color which has 2 states:
 - dark green: scripts injection
 - light green: scripts execution

I'll try to see if I can reproduce the issue. This might be due to an iframe that would be unloaded/reloaded when injecting the content scripts.
